### PR TITLE
fix(tsc-wrapped): add metadata for `type` declarations

### PR DESCRIPTION
**Pull request by alan-agius4**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
At the moment types don't have metadata, and is resulting in incorrect metadata in json when the file only contain `type`

Issue: #18675  

## What is the new behavior?
Type have now metadata

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: packages/tsc-wrapped/src/collector.ts [hunk 1]
```diff
@@ -390,6 +391,17 @@ export class MetadataCollector {
           // Otherwise don't record metadata for the class.
           break;
 
+        case ts.SyntaxKind.TypeAliasDeclaration:
+          const typeDeclaration = <ts.TypeAliasDeclaration>node;
+          if (typeDeclaration.name && isExported(typeDeclaration)) {
+            const name = exportedName(typeDeclaration);
+            if (name) {
+              if (!metadata) metadata = {};
+              metadata[name] = {__symbolic: 'type'};
```
Comment (chuckjaz) [inline packages/tsc-wrapped/src/collector.ts]: Use `{ __symbolic: 'interface' }` instead.  The metadata uses "interface" as the symbolic token value for all non-class types.

**Review by chuckjaz — CHANGES_REQUESTED** [review]

**Comment by chuckjaz** [comment]
Please also squash both the commits into one.

**Comment by alan-agius4** [comment]
@chuckjaz squashed commits and changed the `__symbolic ` value as suggested

File: packages/tsc-wrapped/src/collector.ts (hunk 1 above)
Comment (alan-agius4) [inline packages/tsc-wrapped/src/collector.ts]: done

**Review by chuckjaz — APPROVED** [review]

**Comment by alan-agius4** [comment]
@chuckjaz wasn't this also supposed to be merged as a patch?

**Comment by alan-agius4** [comment]
@chuckjaz & @mhegazy wasn't this also supposed to be merged as a patch?

**Pull request closed without merging**